src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -418,6 +421,16 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         super.execute();
     }
 
+    private File getRootBasedir(MavenProject mavenProject) {
+        if (mavenProject.getParent() == null || 
+            mavenProject.getParent().getBasedir() == null ||
+            !mavenProject.getBasedir().getAbsolutePath()
+                         .contains(mavenProject.getParent().getBasedir().getAbsolutePath())) {
+            return mavenProject.getBasedir();
+        }
+        return getRootBasedir(mavenProject.getParent());
+    }
```

Comment:
this does not appear to be correct. There is no guarantee that a parent in of any submodule in a build is in the reactor, or parent structure.  

For more reading check the [difference between an aggregator project and a parent](https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Project_Inheritance_vs_Project_Aggregation).  It is possible to be both an aggregator and a parent, but it is not required.

Reply:
you could look at the current maven session and what modules are in it. 
or you could possibly use the `maven.multiModuleProjectDirectory` property.